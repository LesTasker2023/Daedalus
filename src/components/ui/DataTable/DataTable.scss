// DataTable SCSS Enhancements
// Import design system first
@import "../../../styles/design-system.css";

.datatable {
  // Only add what Tailwind/design system can't do:

  // Complex table scrolling with design tokens
  &__scroll-container {
    scrollbar-width: thin;
    scrollbar-color: hsl(var(--color-muted)) transparent;

    &::-webkit-scrollbar {
      height: 8px;
      width: 8px;
    }

    &::-webkit-scrollbar-track {
      background: hsl(var(--color-muted) / 0.1);
      border-radius: var(--radius);
    }

    &::-webkit-scrollbar-thumb {
      background-color: hsl(var(--color-muted));
      border-radius: var(--radius);

      &:hover {
        background-color: hsl(var(--color-muted-foreground));
      }
    }
  }

  // Advanced loading animation using design tokens
  &__loading-overlay {
    background: linear-gradient(90deg, transparent, hsl(var(--color-muted) / 0.1), transparent);
    animation: shimmer 2s infinite;
  }

  // Enhanced row hover effects
  &__row {
    transition: background-color var(--transition-default);

    &:hover {
      background-color: hsl(var(--color-accent) / 0.05);
    }

    &--selected {
      background-color: hsl(var(--color-accent) / 0.1);
      border-color: hsl(var(--color-ring));
    }
  }

  // Sort indicator animations
  &__sort-indicator {
    transition: transform var(--transition-fast);

    &--ascending {
      transform: rotate(0deg);
    }

    &--descending {
      transform: rotate(180deg);
    }
  }
}

@keyframes shimmer {
  0% {
    transform: translateX(-100%);
  }
  100% {
    transform: translateX(100%);
  }
}

// Never duplicate Tailwind or design system functionality
