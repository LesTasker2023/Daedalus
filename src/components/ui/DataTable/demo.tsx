'use client';

import { DataTable } from '.';

export default function DataTableDemo() {
  const columns = [
    {
      key: 'name',
      accessorKey: 'name',
      header: 'Name',
    },
    {
      key: 'email',
      accessorKey: 'email',
      header: 'Email',
    },
    {
      key: 'role',
      accessorKey: 'role',
      header: 'Role',
    },
    {
      key: 'status',
      accessorKey: 'status',
      header: 'Status',
    },
  ];

  const data = [
    { id: 1, name: 'John Doe', email: 'john@example.com', role: 'Admin', status: 'Active' },
    { id: 2, name: 'Jane Smith', email: 'jane@example.com', role: 'User', status: 'Active' },
    { id: 3, name: 'Bob Johnson', email: 'bob@example.com', role: 'Editor', status: 'Inactive' },
    { id: 4, name: 'Alice Brown', email: 'alice@example.com', role: 'User', status: 'Active' },
  ];

  return (
    <div className="space-y-6 p-6 max-w-4xl mx-auto">
      <div className="space-y-2">
        <h1 className="text-2xl font-bold">DataTable Component</h1>
        <p className="text-muted-foreground">Advanced data tables with sorting and filtering</p>
      </div>

      <div className="space-y-4">
        <div className="space-y-2">
          <h3 className="text-lg font-semibold">Basic Usage</h3>
          <div className="border rounded-lg p-4 bg-background">
            <DataTable columns={columns} data={data} />
          </div>
        </div>

        <div className="space-y-2">
          <h3 className="text-lg font-semibold">With Pagination</h3>
          <div className="border rounded-lg p-4 bg-background">
            <DataTable columns={columns} data={data} />
          </div>
        </div>
      </div>
    </div>
  );
}
