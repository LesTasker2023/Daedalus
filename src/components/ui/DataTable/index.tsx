export { DataTable, dataTableVariants as variants } from './DataTable';
export type { DataTableProps } from './DataTable';
export { default as DataTableDemo } from './demo';
