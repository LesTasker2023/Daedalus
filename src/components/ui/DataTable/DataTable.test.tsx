import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';
import { DataTable } from './DataTable';

describe('DataTable', () => {
  const mockData = [
    { id: 1, name: 'Test Item 1', value: 'A' },
    { id: 2, name: 'Test Item 2', value: 'B' },
  ];

  const mockColumns = [
    { key: 'name', header: 'Name' },
    { key: 'value', header: 'Value' },
  ];

  const renderBasicDataTable = (props = {}) => {
    return render(
      <DataTable data-testid="datatable" data={mockData} columns={mockColumns} {...props} />
    );
  };

  describe('Snapshots', () => {
    it('matches default snapshot', () => {
      const { container } = renderBasicDataTable();
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches loading state snapshot', () => {
      const { container } = renderBasicDataTable({ loading: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches disabled state snapshot', () => {
      const { container } = renderBasicDataTable({ disabled: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches error state snapshot', () => {
      const { container } = renderBasicDataTable({ error: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches required state snapshot', () => {
      const { container } = renderBasicDataTable({ required: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches checked state snapshot', () => {
      const { container } = renderBasicDataTable({ checked: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches selected state snapshot', () => {
      const { container } = renderBasicDataTable({ selected: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches hover state snapshot', () => {
      const { container } = renderBasicDataTable({ hover: true });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Basic Functionality', () => {
    it('renders correctly', () => {
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });

  describe('States', () => {
    it('handles loading state correctly', () => {
      renderBasicDataTable({ loading: true });
      const element = document.querySelector('.datatable.state-loading');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for loading state
    });
    it('handles disabled state correctly', () => {
      renderBasicDataTable({ disabled: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for disabled state
    });
    it('handles error state correctly', () => {
      renderBasicDataTable({ error: true });
      const element = document.querySelector('.datatable.state-error');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for error state
    });
    it('handles required state correctly', () => {
      renderBasicDataTable({ required: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for required state
    });
    it('handles checked state correctly', () => {
      renderBasicDataTable({ checked: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for checked state
    });
    it('handles selected state correctly', () => {
      renderBasicDataTable({ selected: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for selected state
    });
    it('handles hover state correctly', () => {
      renderBasicDataTable({ hover: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific assertions for hover state
    });
  });

  describe('Events', () => {
    it('handles onSort correctly', async () => {
      const onSort = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onSort });

      // TODO: Add specific event triggering based on onSort
      expect(onSort).toBeDefined();
    });
    it('handles onFilter correctly', async () => {
      const onFilter = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onFilter });

      // TODO: Add specific event triggering based on onFilter
      expect(onFilter).toBeDefined();
    });
    it('handles onPageChange correctly', async () => {
      const onPageChange = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onPageChange });

      // TODO: Add specific event triggering based on onPageChange
      expect(onPageChange).toBeDefined();
    });
    it('handles onRowClick correctly', async () => {
      const onRowClick = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onRowClick });

      // TODO: Add specific event triggering based on onRowClick
      expect(onRowClick).toBeDefined();
    });
    it('handles onRowDoubleClick correctly', async () => {
      const onRowDoubleClick = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onRowDoubleClick });

      // TODO: Add specific event triggering based on onRowDoubleClick
      expect(onRowDoubleClick).toBeDefined();
    });
  });

  describe('Accessibility', () => {
    it.skip('can be focused - SKIPPED: Non-focusable element', () => {
      // This element cannot receive focus (div/span/table)
      // Focus tests disabled for accessibility accuracy
      expect(true).toBe(true);
    });

    it('has proper ARIA attributes', () => {
      renderBasicDataTable();
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
      // TODO: Add specific ARIA attribute tests based on component type
    });

    it.skip('supports keyboard navigation - SKIPPED: Non-focusable element', () => {
      // This element cannot receive focus (div/span/table)
      // Focus tests disabled for accessibility accuracy
      expect(true).toBe(true);
    });

    it('announces changes to screen readers', () => {
      renderBasicDataTable();
      // TODO: Add screen reader announcement tests
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('respects reduced motion preferences', () => {
      renderBasicDataTable();
      // TODO: Add reduced motion tests
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });

  describe('Custom Styling and Props', () => {
    it('accepts custom className', () => {
      renderBasicDataTable({ className: 'custom-class' });
      const element = screen.getByTestId('datatable');
      expect(element).toHaveClass('custom-class');
    });

    it('forwards refs correctly', () => {
      const ref = vi.fn();
      renderBasicDataTable({ ref });
      // Ref forwarding test - environment dependent
      // expect(ref).toHaveBeenCalledWith(expect.any(HTMLElement));
    });

    it('spreads additional props', () => {
      renderBasicDataTable({ 'data-custom': 'test-value' });
      const element = screen.getByTestId('datatable');
      expect(element).toHaveAttribute('data-custom', 'test-value');
    });
  });

  describe('Edge Cases', () => {
    it('handles undefined props gracefully', () => {
      renderBasicDataTable({ children: undefined });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles null props gracefully', () => {
      renderBasicDataTable({ children: null });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles empty string props', () => {
      renderBasicDataTable({ className: '' });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles rapid prop changes', () => {
      const { rerender } = renderBasicDataTable({ className: 'class1' });
      rerender(
        <DataTable
          data-testid="datatable"
          data={mockData}
          columns={mockColumns}
          className="class2"
        />
      );
      const element = screen.getByTestId('datatable');
      expect(element).toHaveClass('class2');
    });

    it('handles complex nested content', () => {
      render(
        <DataTable data-testid="datatable" data={mockData} columns={mockColumns}>
          <div>
            <span>Nested content</span>
            <p>More content</p>
          </div>
        </DataTable>
      );
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('maintains functionality with many children', () => {
      render(
        <DataTable data-testid="datatable" data={mockData} columns={mockColumns}>
          {Array.from({ length: 100 }, (_, i) => (
            <div key={i}>Item {i}</div>
          ))}
        </DataTable>
      );
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles component unmounting cleanly', () => {
      const { unmount } = renderBasicDataTable();
      expect(() => unmount()).not.toThrow();
    });

    it('preserves functionality after remounting', () => {
      const { unmount } = renderBasicDataTable();
      unmount();
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });
});

// TODO: Review and customize generated tests based on component-specific requirements
// TODO: Add component-specific interaction tests
// TODO: Verify all variant combinations work correctly
// TODO: Test integration with form libraries if applicable
// TODO: Add performance tests for complex components
