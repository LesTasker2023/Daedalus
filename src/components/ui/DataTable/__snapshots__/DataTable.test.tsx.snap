// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable > Snapshots > matches checked state snapshot 1`] = `
<div
  aria-disabled="false"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches default snapshot 1`] = `
<div
  aria-disabled="false"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches disabled state snapshot 1`] = `
<div
  aria-disabled="true"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm disabled:opacity-50 disabled:pointer-events-none"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50 pointer-events-none"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50 pointer-events-none"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches error state snapshot 1`] = `
<div
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm state-error"
>
  <div
    class="p-4 text-center"
  >
    <div
      class="text-destructive font-medium mb-2"
    >
      Error
    </div>
    <div
      class="error-message text-sm text-muted-foreground"
    />
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches hover state snapshot 1`] = `
<div
  aria-disabled="false"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches loading state snapshot 1`] = `
<div
  class="datatable state-loading flex items-center justify-center p-8 min-h-[200px]"
>
  <div
    class="flex items-center space-x-2"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-loader-circle h-6 w-6 animate-spin spinner"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
    <span
      class="text-muted-foreground"
    >
      Loading table data...
    </span>
    <span
      class="sr-only"
      role="status"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches required state snapshot 1`] = `
<div
  aria-disabled="false"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
  required=""
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches selected state snapshot 1`] = `
<div
  aria-disabled="false"
  class="datatable w-full relative datatable--default border border-border datatable--size-default text-sm"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="datatable__scroll-container overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
      role="table"
    >
      <thead
        class="[&_tr]:border-b"
      >
        <tr>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="h-10 px-2 text-left align-middle font-medium text-muted-foreground"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="flex items-center space-x-1"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      >
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="datatable__row border-b transition-colors hover:bg-muted/50"
          role="row"
        >
          <td
            class="p-2 align-middle"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="p-2 align-middle"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
